### PR TITLE
Add metrics subsystem

### PR DESCRIPTION
**Pull request by waitinfuture**

### CODE REVIEW

File: dev/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<!DOCTYPE module PUBLIC
```
Comment (TanYuxin-tyx) [inline dev/checkstyle.xml:1]: Missing license

File: pom.xml [hunk 2]
```diff
@@ -35,6 +35,17 @@
     <maven.compiler.source>${java.version}</maven.compiler.source>
     <maven.compiler.target>${java.version}</maven.compiler.target>
     <maven.version>3.5.4</maven.version>
+    <build.testJarPhase>prepare-package</build.testJarPhase>
+    <CodeCacheSize>512m</CodeCacheSize>
```
Comment (TanYuxin-tyx) [inline pom.xml:39]: Where is CodeCacheSize used? I don't get it.

File: pom.xml [hunk 3]
```diff
@@ -35,6 +35,17 @@
...
+    <scala.version>2.12.10</scala.version>
+    <scala.binary.version>2.12</scala.binary.version>
+    <slf4j.version>1.7.16</slf4j.version>
+    <guava.version>14.0.1</guava.version>
+    <codahale.metrics.version>3.2.6</codahale.metrics.version>
```
Comment (TanYuxin-tyx) [inline pom.xml:44]: 3.2.6 is a very old version and why do we use this version? The new version is 4.2.10
Ref: https://github.com/dropwizard/metrics/releases?page=1
https://metrics.dropwizard.io/4.2.0/getting-started.html

**Comment by TanYuxin-tyx** [comment]
Thanks @waitinfuture for the PR.👍🏻  I have some minor comments and please have a look when you are free.

**Comment by waitinfuture** [comment]
> Thanks @waitinfuture for the PR.👍🏻 I have some minor comments and please have a look when you are free.

Hi @TanYuxin-tyx  I have addressed the comments, please take a look. Thanks!

**Pull request merged**